great_expectations/expectations/metrics/map_metric_provider.py

```diff
@@ -1781,17 +1783,41 @@ def _pandas_map_condition_index(
 
     column_name: Union[str, quoted_name]
 
-    if "unexpected_index_column_names" in result_format:
-        unexpected_index_list: Optional[List[Dict[str, Any]]] = []
-        unexpected_index_column_names: List[str] = result_format[
-            "unexpected_index_column_names"
-        ]
-        # column the expectation is being run on
-        expectation_domain_column_name: str = domain_kwargs["column"]
+    expectation_domain_column_name: str = domain_kwargs.get("column")
 
-        unexpected_indices: List[int] = list(df.index)
+    # one named index
+    if df.index.name is not None:
```

Comment:
This function is rather long. Could you factor out this if/else block into it's own function? It looks like the purpose of this bock is to compute the `unexpected_index_list`,  so something like:
```
unexpected_index_list = computed_unexpected_panda_indexes(df, result_format, ...)
```

Reply:
Great idea. I'll add the additional layer of abstraction